panther_analysis_tool/command/benchmark.py

```diff
@@ -0,0 +1,281 @@
+    for _ in range(0, args.iterations):
+        replay_response = backend.run_perf_test(
+            PerfTestParams(
+                zip_bytes=buffer.read(),
+                log_type=log_type,
+                hour=hour_or_err,
+                timeout=timeout.astimezone(),
+            )
+        )
+        if replay_response.data.state == "CANCELED":
+            if len(iterations) == 0:
+                log_extreme_timeout(args, hour_or_err, now)
+                logged = True
+            break
+        if replay_response.data.state in ["ERROR_EVALUATION", "ERROR_COMPUTATION"]:
+            log_error(args, now)
+            if len(iterations) == 0:
+                logged = True
+            break
+        iterations.append(
+            PerformanceTestIteration(
+                read_time_nanos=replay_response.data.replay_summary.read_time_nanos,
+                processing_time_nanos=replay_response.data.replay_summary.processing_time_nanos,
+            )
+        )
+
+    if not logged:
+        log_output(args, hour_or_err, iterations, rule_or_err, now)
+
+    return 0, ""
+
+
+def validate_rule_count(analyses: List[ClassifiedAnalysis]) -> (Union[ClassifiedAnalysis, str]):
+    if len(analyses) != 1:
+        return (
+            f"Only 1 detection must be specified for benchmarking, {len(analyses)} were specified:"
+            f" {[a.file_name for a in analyses]}"
+        )
+    analysis = analyses[0]
+    if analysis.analysis_spec["AnalysisType"] != AnalysisTypes.RULE:
+        return (
+            f"Only rules are supported for performance testing, but {analysis.analysis_spec['AnalysisType']}"
+            f" was provided"
+        )
+    return analysis
+
+
+def validate_log_type(
+    args: argparse.Namespace, rule: ClassifiedAnalysis
+) -> Tuple[Optional[str], Optional[str]]:
+    log_type = getattr(args, "log_type", None)
+    rule_log_types = rule.analysis_spec.get("LogTypes", [])
+    if log_type is None:
+        if len(rule_log_types) > 1:
+            return (
+                log_type,
+                f"Multiple log types specified for {rule.file_name}: {rule_log_types}, please use the"
+                f" --log-type arg to specify one.",
+            )
+        log_type = rule_log_types[0]
+    else:
+        if not str(log_type).casefold() in map(str.casefold, rule_log_types):
+            return (
+                log_type,
+                f"Provided log type {log_type} was not found in log types for {rule.file_name}:"
+                f" {rule_log_types}",
+            )
+    return log_type, None
+
+
+def validate_hour(
+    args: argparse.Namespace, log_type: str, backend: BackendClient
+) -> Union[datetime.datetime, str]:
+    hour = getattr(args, "hour", None)
+    now_truncated = datetime.datetime.now().replace(
+        minute=0, second=0, microsecond=0
+    ).astimezone() + datetime.timedelta(microseconds=-1)
+    window_begin = now_truncated + datetime.timedelta(weeks=-2, microseconds=1)
```

Comment:
hmm I wonder if we should try and get this from the backend since this number can change and has changed over the last few months.

Reply:
Makes sense - do you know if it's currently retrievable?

Reply:
I think the UI gets it via an api call, but yeah to get it might require a new protected api endpoint

Reply:
Recorded under M2: https://app.asana.com/0/1205095847919624/1205187529819686/f

Reply:
sweet thanks